google/cloud/aiplatform/experimental/vertex_model/serializers.py

```diff
@@ -0,0 +1,150 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from concurrent import futures
+import datetime
+import functools
+import inspect
+import logging
+import pandas as pd
+import sys
+import threading
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Iterable,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    Union,
+)
+
+import proto
+import torch 
+
+from google.api_core import operation
+from google.auth import credentials as auth_credentials
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform import utils
+from google.cloud.aiplatform.compat.types import encryption_spec as gca_encryption_spec
+from google.cloud import aiplatform
+                         
+from torch.utils.data import Dataset, Dataloader
+
+def _serialize_dataframe(artifact_uri, obj: pd.DataFrame, 
+                                 temp_dir: str, dataset_type: str):
+        # Designate csv path and write the pandas DataFrame to the path
+        # Convention: file name is my_training_dataset, my_test_dataset, etc.
+        path_to_csv = temp_dir + "/" + "my_" + dataset_type + "_dataset.csv"
+        obj.to_csv(path_to_csv)
+
+        gcs_bucket, gcs_blob_prefix = extract_bucket_and_prefix_from_gcs_path(artifact_uri)
+
+        local_file_name = pathlib.Path(path_to_csv).name
+        blob_path = local_file_name
+
+        if gcs_blob_prefix:
+            blob_path = "/".join([gcs_blob_prefix, blob_path])
+
+        client = storage.Client(project=initializer.global_config.project, 
+                                credentials=initializer.global_config.credentials)
+
+        bucket = client.bucket(gcs_bucket)
+        blob = bucket.blob(blob_path)
+        blob.upload_from_filename(path_to_csv)
+
+        gcs_path = "".join(["gs://", "/".join([blob.bucket.name, blob.name])])
+        return gcs_path
+
+def _deserialize_dataframe(cls, artifact_uri):
+    """ Provides out-of-the-box deserialization after training and prediction is complete """
+    
+    gcs_bucket, gcs_blob = utils.extract_bucket_and_prefix_from_gcs_path(
+        artifact_uri
+    )
+
+    client = storage.Client(project=initializer.global_config.project, 
+                            credentials=initializer.global_config.credentials)
+    bucket = client.bucket(gcs_bucket)
+    blob = bucket.blob(gcs_blob)
+
+    # Incrementally download the CSV file until the header is retrieved
+    first_new_line_index = -1
+    start_index = 0
+    increment = 1000
+    line = ""
+
+    try:
+        logger = logging.getLogger("google.resumable_media._helpers")
+        logging_warning_filter = utils.LoggingFilter(logging.INFO)
+        logger.addFilter(logging_warning_filter)
+
+        while first_new_line_index == -1:
```

Comment:
This method is specifically for extracting the header. This should instead copy from GCS and load the entire Dataframe. These methods can be removed for now and added back in a follow up PR.